client/src/pages/reviews/ReviewPaper.tsx

```diff
@@ -0,0 +1,111 @@
+import { Dialog, Typography, Rating, Avatar, Button, Paper, Grid, Box } from '@mui/material';
+import { useState } from 'react';
+import { ReviewViewModelType } from '../../models/ReviewViewModel';
+import { useTheme } from '@mui/material/styles';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { deleteReview } from '../../api/review';
+import { BookViewModelType } from '../../models/BookViewModel';
+import ErrorOutlineIcon from '@mui/icons-material/ErrorOutline';
+import { ApiResponseError, imgUrl } from '../../utils/utils';
+import useAlert from '../../utils/alerts/useAlert';
+import UpdateReviewForm from './UpdateReviewForm';
+import AuthorizedView from '../../components/auth/AuthorizedView';
+import { useAuth } from '../../utils/auth/useAuth';
+
+function ReviewPaper({ review, book }: { review: ReviewViewModelType; book: BookViewModelType }) {
+  const [deleteError, setDeleteError] = useState<string | null>(null);
+  const { handleError } = useAlert();
+  const theme = useTheme();
+  const queryClient = useQueryClient();
+  const { user } = useAuth();
+
+  const [open, setOpen] = useState(false);
+
+  const handleOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const mutation = useMutation({
+    mutationFn: deleteReview,
+    onSuccess: async () => {
+      queryClient.invalidateQueries({ queryKey: ['reviews', book.id] });
+      queryClient.invalidateQueries({ queryKey: ['books', book.id] });
+    },
+    onError: (err) => {
+      if (err instanceof ApiResponseError && err.error.code == 'REVIEW_NOT_FOUND') {
+        setDeleteError('Ta opinia już nie istnieje.');
+      } else {
+        handleError(err);
+      }
+    },
+  });
+  return (
+    <Paper key={review.id} elevation={2} sx={{ padding: 2, width: '75%', mb: 2 }}>
+      <Grid container>
+        <Grid container direction={'row'} wrap="nowrap">
+          <Grid item xs={1} minHeight={56} minWidth={56}>
+            {/* imgUrl(book.coverPictureUrl, '/podstawowa-ksiazka-otwarta.jpg') */}
```

Comment:
komentarz

Reply:
niesamowite bo u mnie lokalnie nie ma tej linijki a jest up to date branch xD